### DISCUSSION TITLE
Push the website to Github Pages

### DISCUSSION THREAD

**Discussion by theadanielskocher** [vuejs/vitepress#3003] [Q&A]
I have finished my website and I want to push my website to Github pages by hand, but nothing works. How can I address this issue?

![image](https://github.com/vuejs/vitepress/assets/88028913/0448a283-d089-486a-aa1f-48acdd7240ae)

Thank you for your answer!

**Comment by brc-dd** [comment]
Please follow the deploy guide - https://vitepress.dev/guide/deploy#github-pages and if any issue arises, create one with proper debugging information including a minimal reproducible example and the steps you followed.

**Comment by theadanielskocher** [comment]
When I try to paste this code: https://ideone.com/FkP3KS to `.github/workflows`, I get this error:
```
Dependencies lock file is not found in /home/runner/work/dang7rickroll.github.io/dang7rickroll.github.io. Supported file patterns: package-lock.json,npm-shrinkwrap.json,yarn.lock
```
How can I address this issue? Thank you so much.

**Reply by brc-dd** [reply]
You need to commit your lock file too. A lock file will be generated when you run `npm install` locally.

**Reply by laplacetw** [reply]
I've committed pnpm-lock.yaml with VitePress official deploy guide and had the same error. So I use the `--no-frozen-lockfile` :
```yml
- name: Install dependencies
        run: pnpm install --no-frozen-lockfile
```

@brc-dd I follow the official deploy guide and deploy to GitHub pages with base "/project-name/". But why when I type "username.github.io/project-name/" into browser then it will become "username.github.io" ?
* [laplacetw.github.io/vitepress-theme-trigger/](https://laplacetw.github.io/vitepress-theme-trigger/)
* [.github/workflows/gh-pages.yml](https://github.com/laplacetw/vitepress-theme-trigger/blob/main/.github/workflows/gh-pages.yml)

**Reply by brc-dd** [reply]
@laplacetw Your pagination component is not SSR friendly and there you're doing `history.pushState({ page: blogStore.currentPage }, 'goPage()', '/')` -- that's why it's going back to / instead of /base/

**Reply by laplacetw** [reply]
@brc-dd really appreciate your help ! I made that dynamic pagination component because I don't like to build .html for every pagination page. I'm not sure which one is better.

**Reply by brc-dd** [reply]
That's fine. You just need to wrap it with `<ClientOnly>` or call that `history` thing inside `onMounted`.  Anyway that history call should have something like `withBase('/')` for it work properly

**Comment by hf-xz** [comment]
Hey, I get the same situation yesterday. You are getting this page (screen shot) because it's build by `jekyll` but not `vitepress`. Jekyll is the default method used by github to build your `github.io` so you need to change this. see [this](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#about-publishing-sources)

> If you want to use a build process other than Jekyll or you do not want a dedicated branch to hold your compiled static files, we recommend that you write a GitHub Actions workflow to publish your site.

Follow [this section](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#publishing-with-a-custom-github-actions-workflow), you can change your build method from jekyll to custom Github Actions workflow.

After change the settings to this:

![image](https://github.com/vuejs/vitepress/assets/27397614/04d97d7a-c201-4025-b225-cb23e400148f)

You may follow [vitepress guide](https://vitepress.dev/guide/deploy#github-pages) to write your own workflow.

The code given by the guide is good enough for me. I just changed two settings and it works great.

- first, at `on-push-branches`, change the 'main' to the 'master' which I used in my repo
- second, at steps `Build with VitePress` and `Upload artifact`, change the path `docs/.vitepress/*` to `.vitepress/*` because the `.vitepress` is on my project root.
